bin/detect

```diff
@@ -1,4 +1,5 @@
 #!/bin/sh
 # bin/detect BUILD_DIR
 
-! which cmake >/dev/null && echo "CMake"
+HAVE_CMAKE=$(which cmake>/dev/null)
+[ -z "$HAVE_CMAKE" ] && echo "CMake"
```

Comment:
What you have done here @andrecavallari, is to make this script always echo `CMake`:

```shell
$ which cmake
/usr/local/bin/cmake

$ which foo
foo not found

$ HAVE_CMAKE=$(which cmake>/dev/null)

$ [ -z "$HAVE_CMAKE" ] && echo "CMake"
CMake

$ HAVE_CMAKE=$(which foo>/dev/null)

$ [ -z "$HAVE_CMAKE" ] && echo "CMake"
CMake
```

Before, it only echoed `CMake` if `cmake` wasn't found:

```shell
$ ! which cmake >/dev/null && echo "CMake"

$ echo $?
1

$ ! which foo >/dev/null && echo "CMake"
CMake
```

I think the point of the `detect` script is to detect if it is necessary to run the buildpack. I.e., if CMake isn't installed, we want to run the buildpack and install it. If `cmake` is present, we don't want to run the buildpack.

Reply:
Yeah, you're right, but it looks like heroku CI started requiring exit with the correct code too, so I rewrote detect according to the [documentation](https://devcenter.heroku.com/articles/buildpack-api#bin-detect) and it's updated in this Pull Request